pysolr.py

```diff
@@ -823,7 +825,17 @@ def delete(self, id=None, q=None, commit=True, waitFlush=None, waitSearcher=None
         elif id is not None and q is not None:
             raise ValueError('You many only specify "id" OR "q", not both.')
         elif id is not None:
-            m = '<delete><id>%s</id></delete>' % id
+            if type(id) not in (list, set):
+                id = [ id ]
+            else:
+                id = filter(lambda x: x is not None, id)
+            if len(id) > 0:
+                m = '<delete>'
```

Comment:
This block could be changed to avoid repeated string concatenation:

`m = '<delete>%s</delete>' % ''.join('<id>%s</id>' % i for i in id)`